### DISCUSSION TITLE
[help] Growing window branching

### DISCUSSION THREAD

**Discussion by dmi3kno** [ropensci/targets#1149] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

I have a grouped dataframe, over which I am dynamically branching. It is time-indexed data, so I am branching over years. I need to compute some metrics based on current and previous data (but not including future data).

In other words, I want to have a target that would have access to its own slice of the data and every previous slice. I use `map()` over the previous target, so my dynamic branching is simple
```r
list(
slice1, 
slice2, 
slice3,
...)
```

What I would like to have is something like this (partially pooled slices)

```r
list(
c(slice1), 
c(slice1, slice2), 
c(slice1, slice2, slice3),
...)
```
This sounds like `head(i)` to me, but I am not sure how to [compose such a pattern](https://books.ropensci.org/targets/dynamic.html#patterns)

**Comment by dmi3kno** [comment]
Here's what I think I need
```r
tar_pattern(
  map(year_target, head(year_target, n=i_year)),
  year_target=10,
  i_year=10
)
#> Error in checkHT(n, d <- dim(x)) :  invalid 'n' - must be numeric, possibly NA.
```

**Comment by wlandau** [comment, marked as answer]
I suggest mapping over a vector of dates and filtering the data on each of those dates.

```r
list(
  tar_target(name = data, command = get_data()),
  tar_target(name = max_date, command = unique(data$date)),
  tar_target(
    name = result,
    command = data %>%
      filter(date <= max_data),
      summarize_subset(),
    pattern = map(max_date)
  )
)
```
